Sources/MusicPlayer/Players/MPRISNowPlaying.swift

```diff
@@ -7,79 +7,54 @@
 //  file, You can obtain one at https://mozilla.org/MPL/2.0/.
 //
 
-#if os(Linux)
-
-import playerctl
+import DBus
+import Foundation
+import MPRIS
 
 extension MusicPlayers {
-    
-    public final class MPRISNowPlaying: NowPlaying {
-        
-        private let manager: UnsafeMutablePointer<PlayerctlPlayerManager>
-        private var signals: [gulong] = []
-        
-        public init?() {
-            guard let manager = playerctl_player_manager_new(nil) else {
-                return nil
+    /// Just like `NowPlaying`, but automatically find available MPRIS players.
+    public final class MPRISNowPlaying: Agent {
```

Comment:
Here we have player selection logic twice, in different repo. How about deriving from `NowPlaying`, only update player list and let `NowPlaying` do the selection?